### ISSUE TITLE
Code owners

### ISSUE THREAD

**Issue by ThePrimeagen** [ThePrimeagen/ts-rust-zig-deez#95]
I have to get this working... unsure when i will, but i will!  Merged pr #90 for this, just never used codeowners as a way to manage the suppository, glad to try

**Comment by novusnota** [comment]
This CODEOWNERS file only starts being meaningful once people/orgs listed in it have write permissions to the repo.

And it's purpose is to manage branch protection rules: people listed there are automatically requested for review when someone opens a pull request that modifies code that they own as specified in it.

P.S.: Also, to decrease overall clutter in the root of the repo, that file could be moved to .github/ folder and it would still operate the same here on GitHub.

**Comment by ThePrimeagen** [comment]
@novusnota is this branch protection stuff automagic?

so if ShyRy is elixerdeez, he cannot merge a change with zig?

**Comment by novusnota** [comment]
@ThePrimeagen yes! But first one have to:
1. Enable branch protection rule for master
2. Check "Require pull requests before merging"
3. Check "Require review from Code Owners"
4. Validate the correctness of CODEOWNERS file ([gitattributes](https://git-scm.com/docs/gitattributes)-like format) — GitHub will tell something like "This CODEOWNERS file is valid" when viewing it from the web interface.

And then anytime anyone touches your Zig nuts, merging would be blocked until you explicitly grant your permission.

But if, say, ShyRy commits something for elixerdeez, GitHub would still require a review, but only from him and any other code owners assigned to elixerdeez in CODEOWNERS file. Self-admission, baby!

**Comment by novusnota** [comment]
Rules in a file work quite simple (like in CSS): successive rules override preceding ones.

Here's an example:

```codeowners
# You here are supreme code owner of everything
* @ThePrimeagen

# ShyRy with elixerdeez
/elixir/ @ryanwinchester
```

ShyRy here is now responsible for elixerdeez, so GitHub won't bother you when he edits his files, but if he starts editing anything else, which isn't /elixir, GitHub would require a code review from the related code owner. If there is none — it would require review from you.

**Comment by novusnota** [comment]
@bhansconnect yes, code owners also need to have write permissions to the repo for that

**Comment by thedanvail** [comment]
I can help merge PRs if I'm understanding the call for help correctly.

**Comment by zivlakmilos** [comment]
Are codeowners setup completed?
I don't see option to merge my own pull requests. Don't know if I doing it correct.

**Comment by vhladko** [comment]
> Is codeowners setup completed? I don't see option to merge my own pull requests. Don't know if I doing it correct.

It's not completed yet, you can tag me for now if your PR is ready to be merged.

**Comment by leo-costa** [comment]
> > Is codeowners setup completed? I don't see option to merge my own pull requests. Don't know if I doing it correct.
> 
> 
> 
> It's not completed yet, you can tag me for now if your PR is ready to be merged.

Any updates on this?
